tests/models/decoder_only/vision_language/test_models.py

```diff
@@ -0,0 +1,588 @@
+    test_info = VLM_TEST_SETTINGS[model_type]
+    vlm_utils.run_embedding_test(
+        test_info=test_info,
+        model=model,
+        max_tokens=max_tokens,
+        num_logprobs=num_logprobs,
+        dtype=dtype,
+        distributed_executor_backend=distributed_executor_backend,
+        size_wrapper=size_wrapper,
+        hf_runner=hf_runner,
+        vllm_runner=vllm_runner,
+        image_assets=image_assets)
+
+
+@pytest.mark.parametrize(
+    "model_type,model,max_tokens,num_logprobs,dtype,distributed_executor_backend,num_frames,size_wrapper",
+    vlm_utils.get_parametrized_options(
+        VLM_TEST_SETTINGS,
+        test_type=VLMTestType.VIDEO,
+        fork_new_process_for_each_test=False,
+    ))
+def test_video_models(model_type: str, model: str, max_tokens: int,
+                      num_logprobs: int, dtype: str,
+                      distributed_executor_backend: Optional[str],
+                      num_frames: int, size_wrapper: ImageSizeWrapper,
+                      hf_runner: Type[HfRunner], vllm_runner: Type[VllmRunner],
+                      video_assets: _VideoAssets):
+    test_info = VLM_TEST_SETTINGS[model_type]
+    vlm_utils.run_video_test(
+        test_info=test_info,
+        model=model,
+        num_frames=num_frames,
+        max_tokens=max_tokens,
+        num_logprobs=num_logprobs,
+        dtype=dtype,
+        distributed_executor_backend=distributed_executor_backend,
+        size_wrapper=size_wrapper,
+        hf_runner=hf_runner,
+        vllm_runner=vllm_runner,
+        video_assets=video_assets,
+    )
+
+
+@pytest.mark.parametrize(
+    "model_type,model,max_tokens,num_logprobs,dtype,distributed_executor_backend,custom_test_opts",
+    vlm_utils.get_parametrized_options(
+        VLM_TEST_SETTINGS,
+        test_type=VLMTestType.CUSTOM_INPUTS,
+        fork_new_process_for_each_test=False,
+    ))
+def test_custom_inputs_models(
+    model_type: str,
+    model: str,
+    max_tokens: int,
+    num_logprobs: int,
+    distributed_executor_backend: Optional[str],
+    dtype: str,
+    custom_test_opts: CustomTestOptions,
+    hf_runner: Type[HfRunner],
+    vllm_runner: Type[VllmRunner],
+):
+    test_info = VLM_TEST_SETTINGS[model_type]
+    vlm_utils.run_custom_inputs_test(
+        test_info=test_info,
+        model=model,
+        max_tokens=max_tokens,
+        num_logprobs=num_logprobs,
+        dtype=dtype,
+        custom_test_opts=custom_test_opts,
+        distributed_executor_backend=distributed_executor_backend,
+        hf_runner=hf_runner,
+        vllm_runner=vllm_runner)
+
+
+#### Tests filtering for things running each test as a new process
+@pytest.mark.parametrize(
+    "model_type,model,max_tokens,num_logprobs,dtype,distributed_executor_backend,size_wrapper",
+    vlm_utils.get_parametrized_options(
+        VLM_TEST_SETTINGS,
+        test_type=VLMTestType.IMAGE,
+        fork_new_process_for_each_test=True,
```

Comment:
Actually, we only need to fork new process for distributed tests (the tests that have `distributed_executor_backend`, so we can simplify this a bit)

Reply:
Doesn't is also get forked for tests that don't set the distributed executor backend that are decorated with things like `@large_gpu_test` (because of [this](https://github.com/vllm-project/vllm/blob/main/tests/utils.py#L590))? Originally, I was thinking the same, but I had written it this way as a separate flag because there were a couple of tests like [GLM-4](https://github.com/vllm-project/vllm/blob/main/tests/models/decoder_only/vision_language/test_glm4.py#L97) using this decorator without tensor parallelism or the distributed executor backend 🤔

Reply:
I think forking in that case is actually unnecessary, just a copy-paste error from me 😅

Reply:
Awesome - I got rid of the extra `fork_new_process_for_each_test ` on the test info objects, and changed the filtering to check against whether or not `distributed_executor_backend is None` matches the bool passed here instead!